common/mariadb/templates/pre-change-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if or .Values.databases .Values.users }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "fullName" . }}-pre-change"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ default "unknown" .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    "helm.sh/hook": pre-upgrade,pre-rollback
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ include "fullName" . }}-pre-change"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ .Values.name }}-db-exec
+      containers:
+      - name: pre-change-job
+        image: "{{ required ".Values.global.dockerHubMirrorAlternateRegion is missing" .Values.global.dockerHubMirrorAlternateRegion }}/{{ .Values.pre_change_hook.image }}:{{ .Values.pre_change_hook.image_version }}"
+        command:
+        - kubectl-v{{ .Values.pre_change_hook.kubectl_version }}
+        - exec
+        - -c
+        - mariadb
+        - deploy/{{ include "fullName" . }}
```

Comment:
Why is this `deploy`? Is that short for `deployment`? If yes, why did you chose the short-form instead of the full name?

Reply:
> Why is this `deploy`? Is that short for `deployment`?

Yes:
```
$ kubectl api-resources | grep deploy
deployments                        deploy           apps/v1                                true         Deployment
```
> If yes, why did you chose the short-form instead of the full name?

That is the way it is done in the [official documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#exec):
> kubectl exec deploy/mydeployment -- date